plugins/org.locationtech.udig.catalog/src/org/locationtech/udig/catalog/internal/CatalogImpl.java

```diff
@@ -1155,10 +1155,12 @@ public int compare( IService o1, IService o2 ) {
                     double metric1 = info1.getMetric();
                     double metric2 = info2.getMetric();
                     
+                    // needs to be sorted in descending order, so that the most matching
```

Comment:
sorting got mixed up (look at the javadoc for IServiceInfo.getMetric(), it says 1, if the request matched all parameters and 0 if not).